packages/@react-aria/select/src/useSelect.ts

```diff
@@ -104,12 +115,31 @@ export function useSelect<T>(props: AriaSelectOptions<T>, state: SelectState<T>,
         triggerProps['aria-label'] && !triggerProps['aria-labelledby'] ? triggerProps.id : null,
         valueId
       ].filter(Boolean).join(' '),
-      onFocus() {
+      onFocus(e: FocusEvent) {
+        if (state.isFocused) {
+          return;
+        }
+
+        if (props.onFocus) {
+          props.onFocus(e);
+        }
+
         state.setFocused(true);
       },
-      onBlur() {
+      onBlur(e: FocusEvent) {
+        if (e.relatedTarget === triggerRef.current || popoverRef.current?.contains(e.relatedTarget as HTMLElement)) {
+          return;
+        }
+
+        if (props.onBlur) {
+          props.onBlur(e);
+        }
+
         state.setFocused(false);
-      }
+      },
+      onFocusChange: props.onFocusChange,
+      onKeyDown: props.onKeyDown,
```

Comment:
well `FocusableProps` has them so I thought they somehow have to be handled - I could certainly not handle them there but in `Picker.tsx` on the `FieldButton` like I do it already for `autoFocus`

Reply:
Not sure how you'd like to proceed with that one, should i keep them here or handle them in Picker.tsx directly - I think they belong there because `AriaSelectProps` has them so devs using `useSelect` most likely expect those are handled but it's you call.